DataFormats/Candidate/interface/LeafCandidate.h

```diff
@@ -33,6 +33,8 @@ namespace reco {
 
     typedef unsigned int index;
 
+    LeafCandidate() {}
```

Comment:
Yes.  The explicit default ctor is needed only because ROOT6 does not recognize the variadic ctor as a default constructor.  I will write a JIRA ticket against ROOT6 for this.  If ROOT6 explicitly knew that this variadic ctor can serve as a default ctor, the explicit default ctor would not be needed.